accept/accept.go

```diff
@@ -0,0 +1,76 @@
+package accept
+
+import (
+	"mime"
+	"net/http"
+	"strings"
+)
+
+const (
+	ALL = "*/*"
+)
+
+var (
+	EventStream = []string{"text/event-stream", "text/*"}
+	HTML        = []string{"text/html"}
+	JSON        = []string{"application/json", "application/javascript"}
+	Plain       = []string{"text/plain"}
+	XML         = []string{"application/xhtml+xml", "application/xml"}
+)
+
+func Accept(mediaTypes []string, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !acceptable(r.Header.Get("Accept"), mediaTypes) {
+			w.WriteHeader(http.StatusNotAcceptable)
+			return
+		}
+		next.ServeHTTP(w, r)
+	})
+}
+
+func AcceptEventStream(next http.Handler) http.Handler {
```

Comment:
Without these helpers it would be a triple accept: `accept.Accept(accept.EventStream, next)`. A bit unsure about the repetition though.